common/src/main/java/com/wynntils/features/user/ItemTextOverlayFeature.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.properties.FeatureInfo;
+import com.wynntils.mc.event.HotbarSlotRenderEvent;
+import com.wynntils.mc.event.SlotRenderEvent;
+import com.wynntils.mc.render.FontRenderer;
+import com.wynntils.wc.custom.item.WynnItemStack;
+import com.wynntils.wc.custom.item.properties.ItemProperty;
+import com.wynntils.wc.custom.item.properties.type.TextOverlayProperty;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@FeatureInfo(category = "Inventory")
+public class ItemTextOverlayFeature extends UserFeature {
+
+    @Config
+    public static boolean powderTierEnabled = true;
+
+    @Config
+    public static boolean powderTierRomanNumerals = true;
+
+    @Config
+    public static FontRenderer.TextShadow powderTierShadow = FontRenderer.TextShadow.OUTLINE;
+
+    @Config
+    public static boolean inventoryTextOverlayEnabled = true;
+
+    @Config
+    public static boolean hotbarTextOverlayEnabled = true;
+
+    @SubscribeEvent
+    public void onRenderSlot(SlotRenderEvent.Post e) {
+        if (!inventoryTextOverlayEnabled) return;
+
+        drawTextOverlay(e.getSlot().getItem(), e.getSlot().x, e.getSlot().y, false);
+    }
+
+    @SubscribeEvent
+    public void onRenderHotbarSlot(HotbarSlotRenderEvent.Post e) {
+        if (!hotbarTextOverlayEnabled) return;
+
+        drawTextOverlay(e.getStack(), e.getX(), e.getY(), true);
+    }
+
+    public void drawTextOverlay(ItemStack item, int slotX, int slotY, boolean hotbar) {
+        if (!(item instanceof WynnItemStack wynnItem)) return;
+        if (!wynnItem.hasProperty(ItemProperty.TEXT_OVERLAY)) return;
+
+        for (TextOverlayProperty overlayProperty : wynnItem.getProperties(ItemProperty.TEXT_OVERLAY)) {
```

Comment:
This for loop... Can we have more than one text overlay per item? If we *really* want to support that, we'd likely have to offset the texts, so as not to have them completely overlap. 

Or, if we in practice only support one text overlay, maybe make that an explicit consistency check, and remove the loop.

Reply:
the placement of text is defined by the properties, so in the (probably rare) case of potentially overlapping texts, we'll just need to take care to give them different offsets

Reply:
also, text overlays that exist in different contexts (inventory vs hotbar) could theoretically be applied to the same item

Reply:
Ok!